examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,211 @@
+                markersize=4 if k == -1 else 1 + 5 * proba_map[ci],
+            )
+    n_clusters_ = len(set(labels)) - (1 if -1 in labels else 0)
+    preamble = "True" if ground_truth else "Estimated"
+    title = f"{preamble} number of clusters: {n_clusters_}"
+    if parameters is not None:
+        parameters_str = ", ".join(f"{k}={v}" for k, v in parameters.items())
+        title += f" | {parameters_str}"
+    ax.set_title(title)
+    plt.tight_layout()
+
+
+# %%
+# Generate sample data
+# --------------------
+# One of the greatest advantages of HDBSCAN over DBSCAN is its out-of-the-box
+# robustness. It's especially remarkable on heterogenous mixtures of data.
+# Like DBSCAN, it can model arbitrary shapes and distributions, however unlike
+# DBSCAN it does not require specification of an arbitray (and indeed tricky)
+# `eps` hyperparameter. For example, below we generate a dataset composed of
+# a mixture of three diagonal Gaussians.
+centers = [[1, 1], [-1, -1], [1.5, -1.5]]
+X, labels_true = make_blobs(
+    n_samples=750, centers=centers, cluster_std=[0.4, 0.1, 0.75], random_state=0
+)
+plot(X, labels=labels_true, ground_truth=True)
+# %%
+# Scale Invariance
+# -----------------
+# It's worth remembering that, while DBSCAN provides a default value for `eps`
+# parameter, it is entirely meaningless and must be tuned for your specific
+# dataset. As a simple demonstration, consider what happens when we find an
+# epsilon value that works for one dataset, and try to apply it to a
+# similar but rescaled versions of the dataset. Below are plots of the original
+# dataset, and versions rescaled by 0.5 and 3 respectively.
+fig, axes = plt.subplots(3, 1, figsize=(10, 12))
+parameters = {"eps": 0.3}
+dbs = DBSCAN(**parameters).fit(X)
+plot(X, dbs.labels_, parameters=parameters, ax=axes[0])
+dbs.fit(0.5 * X)
+plot(0.5 * X, dbs.labels_, parameters=parameters, ax=axes[1])
+dbs.fit(3 * X)
+plot(3 * X, dbs.labels_, parameters=parameters, ax=axes[2])
+
+# %%
+# Indeed, in order to maintain the same results we would have to scale `eps` by
+# the same factor.
+fig, axis = plt.subplots(1, 1, figsize=(12, 5))
+dbs = DBSCAN(eps=0.9).fit(3 * X)
+plot(3 * X, dbs.labels_, parameters={"eps": 0.9}, ax=axis)
+
+# %%
+# While standardizing data (e.g. using
+# :class:`sklearn.preprocessing.StandardScaler`) helps mitigate this problem,
+# great care must be taken to select the appropriate value for `eps`. HDBSCAN
+# is much more robust in this sense. HDBSCAN can be seen as clustering over
+# all possible values of `eps` and extracting the best clusters from all
+# possible clusters (see :ref:`User Guide <HDBSCAN>`). One immediate
+# advantage is that HDBSCAN is scale-invariant.
+fig, axes = plt.subplots(3, 1, figsize=(10, 12))
+hdb = HDBSCAN().fit(X)
+plot(X, hdb.labels_, hdb.probabilities_, ax=axes[0])
+hdb.fit(0.5 * X)
+plot(0.5 * X, hdb.labels_, hdb.probabilities_, ax=axes[1])
+hdb.fit(3 * X)
+plot(3 * X, hdb.labels_, hdb.probabilities_, ax=axes[2])
```

Comment:
Similarly, I think it would also be appropriate having information given in each plot's title so that they are self-explainable.